assets/app/view/game/buy_trains.rb

```diff
@@ -302,7 +302,8 @@ def from_depot(depot_trains, corporation)
             president_assist, _fee = @game.president_assisted_buy(@corporation, train, price)
             entity = @corporation
 
-            if [@corporation, @corporation.owner].include?(@selected_company&.owner) && @step.respond_to?(:ability_timing)
+            if @selected_company && [@corporation, @corporation.owner].include?(@selected_company&.owner) \
```

Comment:
the & after selected_company&.owner is no longer necessary

Reply:
why is this necessary? how can a corporation have no owner?

Reply:
I hadn't noticed that, thanks.  It WAS needed when `@selected_company` could be nil (no private company selected to use its power), and hence trying to call `nil.owner`, rather than a company having no owner.  Now we're short-circuiting earlier for `@selected_company` actually being a company, that can't happen.  I'll tidy that up.

Reply:
I did some more digging, and the reason the logic goes wrong in the first place looks to be that Engine::Player overrides '==', and that override method can return nil if it's being compared to nil, rather than true or false.  I suspect that's confusing include?, and making it think that @corporation.owner == @selected_company&.owner is true, when @selected_company (and hence @selected_company&.owner) is nil.

@corporation.owner == nil definitely returns nil, rather than false.

It feels like '==' should always return true or false, but I'm enough of a noob that I could be *very* wrong on that.

That feels broader than just addressing this particular bug - what's the best way to discuss?

Reply:
we can chat about it on slack if you want